### PR TITLE
error handling

### PR DESCRIPTION
**Pull request by FinnStutzenstein**
Example for error handling. Can be tested when creating a motion (leave everything empty and click 'save') and in the config (go to motions->supporters enter  negative number).

If this is approved, we can use this in every view. The strings in the http service can be adjusted, if wanted.

### CODE REVIEW

**Comment by FinnStutzenstein** [comment]
> It should be included in all occasions, where an error with the http service could happen.
I take this as an OK for this draft. I will edit every view then.

> Also the Errors could be better phrased I think, so they are e.g. coherent with the offline mode.
If you have suggestions for a better string or a missed case, pleas tell. If the offline modus become implemented, we might synchronize both.

**Comment by FinnStutzenstein** [comment]
Please have a look now; I hopfully didn't miss anything.

**Comment by emanuelschuetze** [comment]
Tested: Looks ok for me. Especially the config error is nice. The snackbar error should get more 'error styling' (e.g. red '!' icon). But we can do this polishing later...

File: client/src/app/site/motions/services/change-recommendation-repository.service.ts [hunk 1]
```diff
@@ -49,17 +49,10 @@ export class ChangeRecommendationRepositoryService extends BaseRepository<ViewCh
...
 
     /**
-     * Given a change recommendation view object, a entry in the backend is created and the new
-     * change recommendation view object is returned (as an observable).
-     *
-     * @param {ViewChangeReco} view
-     * @deprecated Will not work with PR #3928. There will just be the id as response to create requests.
-     *  Two possibilities: Make a server change to still retrieve the created object or you have to wait for the
-     *  correct autoupdate.
+     * Given a change recommendation view object, a entry in the backend is created.
```
Comment (jsaalfeld) [inline client/src/app/site/motions/services/change-recommendation-repository.service.ts]: Parameter is missing in the documentation

File: client/src/app/site/users/components/group-list/group-list.component.ts [hunk 2]
```diff
@@ -54,8 +55,14 @@ export class GroupListComponent extends BaseComponent implements OnInit {
      * @param DS The Data Store
      * @param constants Constants
```
Comment (jsaalfeld) [inline client/src/app/site/users/components/group-list/group-list.component.ts]: Parameters are outdated

File: client/src/app/site/users/components/group-list/group-list.component.ts [hunk 3]
```diff
@@ -88,35 +95,39 @@ export class GroupListComponent extends BaseComponent implements OnInit {
      * Saves a newly created group.
      * @param form form data given by the group
```
Comment (jsaalfeld) [inline client/src/app/site/users/components/group-list/group-list.component.ts]: doesn't have a parameter

File: client/src/app/site/users/components/group-list/group-list.component.ts [hunk 4]
```diff
@@ -88,35 +95,39 @@ export class GroupListComponent extends BaseComponent implements OnInit {
...
-        this.cancelEditing();
+        this.repo.create(this.groupForm.value).then(() => {
+            this.groupForm.reset();
+            this.cancelEditing();
+        }, this.raiseError);
     }
 
     /**
      * Saves an edited group.
      * @param form form data given by the group
```
Comment (jsaalfeld) [inline client/src/app/site/users/components/group-list/group-list.component.ts]: doesn't have a parameter

File: client/src/app/site/users/components/user-detail/user-detail.component.ts [hunk 5]
```diff
@@ -56,13 +61,19 @@ export class UserDetailComponent implements OnInit {
      * Constructor for user
      */
```
Comment (jsaalfeld) [inline client/src/app/site/users/components/user-detail/user-detail.component.ts]: the parameters for the constructor are missing

File: client/src/app/site/users/components/user-detail/user-detail.component.ts [hunk 6]
```diff
@@ -89,7 +100,7 @@ export class UserDetailComponent implements OnInit {
      * sets the ownPage variable if the operator owns the page
```
Comment (jsaalfeld) [inline client/src/app/site/users/components/user-detail/user-detail.component.ts]: parameter in documentation is missing

File: client/src/app/site/users/components/user-list/user-list.component.ts [hunk 7]
```diff
@@ -24,13 +25,14 @@ export class UserListComponent extends ListViewBaseComponent<ViewUser> implement
      * @param translate
```
Comment (jsaalfeld) [inline client/src/app/site/users/components/user-list/user-list.component.ts]: the parameters are outdated

File: client/src/app/site/users/components/group-list/group-list.component.ts [hunk 8]
```diff
@@ -133,9 +144,9 @@ export class GroupListComponent extends BaseComponent implements OnInit {
      * @param group
```
Comment (jsaalfeld) [inline client/src/app/site/users/components/group-list/group-list.component.ts]: `group` parameter is outdated, and some documentation would be nice

File: client/src/app/site/users/components/group-list/group-list.component.ts [hunk 9]
```diff
@@ -54,8 +55,14 @@ export class GroupListComponent extends BaseComponent implements OnInit {
...
+        titleService: Title,
+        translate: TranslateService,
+        matSnackBar: MatSnackBar,
+        public repo: GroupRepositoryService,
+        private promptService: PromptService
+    ) {
+        super(titleService, translate, matSnackBar);
     }
 
     public setEditMode(mode: boolean, newGroup: boolean = true): void {
```
Comment (jsaalfeld) [inline client/src/app/site/users/components/group-list/group-list.component.ts]: documentation is missing

File: client/src/app/site/tags/components/tag-list/tag-list.component.ts [hunk 10]
```diff
@@ -126,6 +127,7 @@ export class TagListComponent extends ListViewBaseComponent<ViewTag> implements
             this.cancelEditing();
         }
     }
+
     public keyDownFunction(event: KeyboardEvent): void {
```
Comment (jsaalfeld) [inline client/src/app/site/tags/components/tag-list/tag-list.component.ts]: documentation is missing

File: client/src/app/site/tags/components/tag-list/tag-list.component.ts [hunk 11]
```diff
@@ -37,10 +38,11 @@ export class TagListComponent extends ListViewBaseComponent<ViewTag> implements
     public constructor(
```
Comment (jsaalfeld) [inline client/src/app/site/tags/components/tag-list/tag-list.component.ts]: the documentation is outdated

File: client/src/app/site/motions/components/statute-paragraph-list/statute-paragraph-list.component.ts [hunk 12]
```diff
@@ -122,8 +127,7 @@ export class StatuteParagraphListComponent extends BaseComponent implements OnIn
     public async onDeleteButton(viewStatuteParagraph: ViewStatuteParagraph): Promise<void> {
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/statute-paragraph-list/statute-paragraph-list.component.ts]: parameter is missing in documentation

File: client/src/app/site/motions/components/statute-paragraph-list/statute-paragraph-list.component.ts [hunk 13]
```diff
@@ -109,10 +113,11 @@ export class StatuteParagraphListComponent extends BaseComponent implements OnIn
     /**
      * Saves the statute paragrpah
      */
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/statute-paragraph-list/statute-paragraph-list.component.ts]: parameter is missing in documentation

File: client/src/app/site/motions/components/statute-paragraph-list/statute-paragraph-list.component.ts [hunk 14]
```diff
@@ -85,11 +88,12 @@ export class StatuteParagraphListComponent extends BaseComponent implements OnIn
         }
     }
 
-    public async create(): Promise<void> {
+    public create(): void {
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/statute-paragraph-list/statute-paragraph-list.component.ts]: documentation would be nice

File: client/src/app/site/motions/components/statute-paragraph-list/statute-paragraph-list.component.ts [hunk 15]
```diff
@@ -44,13 +45,15 @@ export class StatuteParagraphListComponent extends BaseComponent implements OnIn
      * @param formBuilder
      */
     public constructor(
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/statute-paragraph-list/statute-paragraph-list.component.ts]: the parameters in documentation are outdated

File: client/src/app/site/motions/components/motion-detail/motion-detail.component.ts [hunk 16]
```diff
@@ -138,6 +137,9 @@ export class MotionDetailComponent extends BaseComponent implements OnInit {
      * @param translate: Translation Service
      */
     public constructor(
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/motion-detail/motion-detail.component.ts]: the parameters in the documentation are outdated

File: client/src/app/site/motions/components/motion-detail-diff/motion-detail-diff.component.ts [hunk 17]
```diff
@@ -48,12 +50,17 @@ export class MotionDetailDiffComponent implements AfterViewInit {
...
         private recoRepo: ChangeRecommendationRepositoryService,
         private dialogService: MatDialog,
         private el: ElementRef
-    ) {}
+    ) {
+        super(title, translate, matSnackBar);
+    }
 
     /**
      * Returns the part of this motion between two change objects
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/motion-detail-diff/motion-detail-diff.component.ts]: return statement is missing in documentation

File: client/src/app/site/motions/components/motion-detail-diff/motion-detail-diff.component.ts [hunk 18]
```diff
@@ -48,12 +50,17 @@ export class MotionDetailDiffComponent implements AfterViewInit {
     public createChangeRecommendation: EventEmitter<LineRange> = new EventEmitter<LineRange>();
 
     public constructor(
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/motion-detail-diff/motion-detail-diff.component.ts]: documentation is missing

File: client/src/app/site/motions/components/motion-detail-diff/motion-detail-diff.component.ts [hunk 19]
```diff
@@ -36,7 +38,7 @@ import {
     templateUrl: './motion-detail-diff.component.html',
     styleUrls: ['./motion-detail-diff.component.scss']
 })
-export class MotionDetailDiffComponent implements AfterViewInit {
+export class MotionDetailDiffComponent extends BaseViewComponent implements AfterViewInit {
     @Input()
     public motion: ViewMotion;
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/motion-detail-diff/motion-detail-diff.component.ts]: documentation for this value and the following values are missing

File: client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts [hunk 20]
```diff
@@ -149,8 +154,7 @@ export class MotionCommentSectionListComponent extends BaseComponent implements
     public async onDeleteButton(viewSection: ViewMotionCommentSection): Promise<void> {
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts]: parameter in documentation is missing

File: client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts [hunk 21]
```diff
@@ -136,10 +140,11 @@ export class MotionCommentSectionListComponent extends BaseComponent implements
     /**
      * Saves the categories
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts]: parameter in the documentation is missing

File: client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts [hunk 22]
```diff
@@ -111,11 +114,12 @@ export class MotionCommentSectionListComponent extends BaseComponent implements
         }
     }
 
-    public async create(): Promise<void> {
+    public create(): void {
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts]: documentation would be nice

File: client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts [hunk 23]
```diff
@@ -49,14 +50,16 @@ export class MotionCommentSectionListComponent extends BaseComponent implements
      * @param formBuilder
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/motion-comment-section-list/motion-comment-section-list.component.ts]: the parameters are outdated

File: client/src/app/site/motions/components/motion-change-recommendation/motion-change-recommendation.component.ts [hunk 24]
```diff
@@ -86,10 +89,15 @@ export class MotionChangeRecommendationComponent {
 
     public constructor(
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/motion-change-recommendation/motion-change-recommendation.component.ts]: documentation is missing

File: client/src/app/site/motions/components/category-list/category-list.component.ts [hunk 25]
```diff
@@ -178,8 +183,7 @@ export class CategoryListComponent extends BaseComponent implements OnInit {
     public async onDeleteButton(viewCategory: ViewCategory): Promise<void> {
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/category-list/category-list.component.ts]: parameter in documentation is missing

File: client/src/app/site/motions/components/category-list/category-list.component.ts [hunk 26]
```diff
@@ -145,7 +145,12 @@ export class CategoryListComponent extends BaseComponent implements OnInit {
      */
     public async onSaveButton(viewCategory: ViewCategory): Promise<void> {
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/category-list/category-list.component.ts]: parameter in documentation is missing

File: client/src/app/site/motions/components/category-list/category-list.component.ts [hunk 27]
```diff
@@ -84,7 +86,6 @@ export class CategoryListComponent extends BaseComponent implements OnInit {
      */
     public keyDownFunction(event: KeyboardEvent, viewCategory?: ViewCategory): void {
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/category-list/category-list.component.ts]: parameters in documentation are missing

File: client/src/app/site/motions/components/category-list/category-list.component.ts [hunk 28]
```diff
@@ -60,13 +61,14 @@ export class CategoryListComponent extends BaseComponent implements OnInit {
      * @param formBuilder
      */
     public constructor(
```
Comment (jsaalfeld) [inline client/src/app/site/motions/components/category-list/category-list.component.ts]: parameters in documentation are outdated

File: client/src/app/site/config/components/config-field/config-field.component.ts [hunk 29]
```diff
@@ -116,19 +116,16 @@ export class ConfigFieldComponent extends BaseComponent implements OnInit {
     /**
      * Updates the this config field.
      */
-    private async update(value: any): Promise<void> {
+    private update(value: any): void {
```
Comment (jsaalfeld) [inline client/src/app/site/config/components/config-field/config-field.component.ts]: parameter is missing in the documentation

File: client/src/app/site/base/list-view-base.ts [hunk 30]
```diff
@@ -34,8 +34,8 @@ export abstract class ListViewBaseComponent<V extends BaseViewModel> extends Bas
      * @param titleService the title serivce
      * @param translate the translate service
      */
```
Comment (jsaalfeld) [inline client/src/app/site/base/list-view-base.ts]: documentation is missing the new parameter

File: client/src/app/site/assignments/assignment-list/assignment-list.component.ts [hunk 31]
```diff
@@ -22,8 +23,13 @@ export class AssignmentListComponent extends ListViewBaseComponent<ViewAssignmen
      * @param titleService
      * @param translate
      */
-    public constructor(private repo: AssignmentRepositoryService, titleService: Title, translate: TranslateService) {
-        super(titleService, translate);
+    public constructor(
```
Comment (jsaalfeld) [inline client/src/app/site/assignments/assignment-list/assignment-list.component.ts]: documentation parameters are outdated

File: client/src/app/site/agenda/agenda-list/agenda-list.component.ts [hunk 32]
```diff
@@ -25,10 +26,11 @@ export class AgendaListComponent extends ListViewBaseComponent<ViewItem> impleme
     public constructor(
```
Comment (jsaalfeld) [inline client/src/app/site/agenda/agenda-list/agenda-list.component.ts]: some parameters are missing in the documentation

File: client/src/app/core/services/http.service.ts [hunk 33]
```diff
@@ -18,30 +19,81 @@ export enum HTTPMethod {
...
-            console.log("error", e);
-            throw e;
+            throw this.handleError(e);
+        }
+    }
+
+    /**
+     * Takes an error thrown by the HttpClient. Processes it to return a string that can
+     * be presented to the user.
+     * @param e The error thrown.
```
Comment (jsaalfeld) [inline client/src/app/core/services/http.service.ts]: retrurn value is missing in the documentation

File: client/src/app/core/services/http.service.ts [hunk 34]
```diff
@@ -18,30 +19,81 @@ export enum HTTPMethod {
 @Injectable({
     providedIn: 'root'
 })
-
 export class HttpService {
     /**
      * Construct a HttpService
      *
      * @param http The HTTP Client
```
Comment (jsaalfeld) [inline client/src/app/core/services/http.service.ts]: new parameter is missing in the documentation

File: client/src/app/core/services/http.service.ts [hunk 35]
```diff
@@ -18,30 +19,81 @@ export enum HTTPMethod {
...
+                        const capitalizedKey = key.charAt(0).toUpperCase() + key.slice(1);
+                        return this.translate.instant(capitalizedKey) + ': ' + this.processErrorTexts(e.error[key]);
+                    })
+                    .join(', ');
+            }
+        } else if (e.status === 500) {
+            error += this.translate.instant('A server error occured. Please contact your system administrator.');
+        } else if (e.status > 500) {
+            error += this.translate
+                .instant('The server cound not be reached (%%code%%).')
```
Comment (tsiegleauq) [inline client/src/app/core/services/http.service.ts]: The TranslateParser understands parameters. Replacing this is not necessary.

from the documentation of ngx translate:

>instant(key: string|Array<string>, interpolateParams?: Object): string|Object: Gets the instant translated value of a key (or an array of keys). /!\ This method is synchronous and the default file loader is asynchronous. You are responsible for knowing when your translations have been loaded and it is safe to use this method. If you are not sure then you should use the get method instead.

File: client/src/app/site/motions/components/motion-change-recommendation/motion-change-recommendation.component.html [hunk 36]
```diff
@@ -15,7 +15,7 @@ <h3>Change from line {{ lineRange.from }} to {{ lineRange.to }}:</h3>
     </form>
 </mat-dialog-content>
 <mat-dialog-actions>
-    <button mat-button mat-dialog-close>Abort</button>
+    <button mat-button mat-dialog-close translate>Abort</button>
```
Comment (tsiegleauq) [inline client/src/app/site/motions/components/motion-change-recommendation/motion-change-recommendation.component.html]: I would prefer "Cancel" over "Abort".

File: client/src/app/core/services/http.service.ts (hunk 35 above)
Comment (FinnStutzenstein) [reply client/src/app/core/services/http.service.ts]: `this.translate.instant('The server cound not be reached ({{ code }}).', {code: e.status});` would be fine, but does not work :(
I'll create an issue for this and leave the replace there..

File: client/src/app/site/motions/components/motion-change-recommendation/motion-change-recommendation.component.html (hunk 36 above)
Comment (FinnStutzenstein) [reply client/src/app/site/motions/components/motion-change-recommendation/motion-change-recommendation.component.html]: I think @emanuelschuetze done this in the translation PR (#3989 )

Comment (tsiegleauq) [reply client/src/app/site/motions/components/motion-change-recommendation/motion-change-recommendation.component.html]: another pull request by @emanuelschuetze changed the label here.
Furthermore, `translation` somehow do not work on "button mat-button".
You need to put a <span> inside

**Pull request merged**